Service-Broker-Setup.md

```diff
@@ -0,0 +1,186 @@
+```
+
+Retrieve key `internal-ca-cert` of the CF secret `secret` and store
+the base64-decoded result in the environment variable `CF_CA_CERT`.
+
+```
+CF_CA_CERT="$(kubectl get secret secret --namespace ${CF_NAMESPACE} -o jsonpath="{.data['internal-ca-cert']}" | base64 --decode -)"
+```
+
+Retrieve the administrative password for the CAP cluster and store it
+in the environment variable `CLUSTER_ADMIN_PASSWORD`.
+
+```
+CLUSTER_ADMIN_PASSWORD="$(kubectl get secret secret --namespace ${CF_NAMESPACE} -o jsonpath="{.data['cluster-admin-password']}" | base64 --decode -)"
+```
+
+## Deployment
+
+We can now deploy the broker's chart via
+
+```
+helm install ${CHART} \
+    --namespace ${NAMESPACE} \
+    --set "env.SERVICE_LOCATION=http://cf-usb-sidecar-mysql.${NAMESPACE}.svc.cluster.local:8081" \
+    \
+    --set "env.SERVICE_MYSQL_HOST=${DBHOST}" \
+    --set "env.SERVICE_MYSQL_PORT=${DBPORT}" \
+    --set "env.SERVICE_MYSQL_USER=${DBUSER}" \
+    --set "env.SERVICE_MYSQL_PASS=${DBPASS}" \
+    \
+    --set "env.CF_ADMIN_USER=admin" \
+    --set "env.CF_ADMIN_PASSWORD=${CLUSTER_ADMIN_PASSWORD}" \
+    --set "env.CF_DOMAIN=${DOMAIN}" \
+    --set "env.CF_CA_CERT=${CF_CA_CERT}" \
+    --set "env.UAA_CA_CERT=${UAA_CA_CERT}" \
+    \
+    --set "kube.organization=${DOCKER_ORGANIZATION}" \
+    --set "kube.registry.hostname=${DOCKER_REPOSITORY}"
+```
+
+Notes: 
+
+* The first `--set` tells the new broker where itself will be visible
+  on the internal network of the CAP cluster. Its setup errand
+  provides this information to the USB component of the CAP cluster,
+  so that it can talk to the new broker.
+
+* The remaining assignments provide the connection information for the
+  database and CAP to the broker, as well as the origin information
+  for the docker images referenced by the chart.
+
+
+# Appendix I: Postgres
+
+Deploying the postgres broker and chart is virtually identical to
+deploying mysql. The only differences are:
+
+* The chart variables have prefix `SERVICE_POSTGRESQL_` instead of
+  `SERVICE_MYSQL_`.
+
+* The `SERVICE_LOCATION uses `cf-usb-sidecar-postgres`.
+
+* An additional assignment of the form
+  `--set "env.SERVICE_POSTGRESQL_SSLMODE=disable"` may be required,
+  depending on the setup of the database. See also
+  [Appendix II: Chart variables](#appendix-ii-chart-variables) for
+  more information on the available values.
+
+# Appendix II: Chart variables
+
+|Variable			|Meaning|
+|---				|---|
+|env.SERVICE_LOCATION		|Broker location as seen by CAP cluster|
+|env.SERVICE_(db)_HOST		|Host the database lives on|
+|env.SERVICE_(db)_PORT		|Port the database listens on|
+|env.SERVICE_(db)_USER		|User name for database connections|
+|env.SERVICE_(db)_PASS		|Password for database connections|
+|env.SERVICE_POSTGRESQL_SSLMODE	|Connection to postgres server, one of `disable`, `require`, `verify-ca`, `verify-full`|
```

Comment:
Actually, maybe we should just link to https://godoc.org/github.com/lib/pq?

Reply:
+1, will do.

Reply:
Addressed with commit 0293cbde36258e3f6070d265af2180394c260697